### PR TITLE
Standardize usage of `FileProperty`

### PR DESCRIPTION
**Pull request by marcenacp**

### CODE REVIEW

**Review by ccl-core — APPROVED** [review]
Thanks!

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/join.py [hunk 1]
```diff
@@ -3,52 +3,67 @@
...
-                f'Column "{right_key}" does not exist in node "{right.uid}".'
-                f" Existing columns: {df_right.columns}"
-            )
-            df_left[left_key] = df_left[left_key].transform(
-                apply_transforms_fn, source=left
-            )
-            return df_left.merge(
-                df_right, left_on=left_key, right_on=right_key, how="left"
-            )
+        if len(args) == 2:
```
[the remark this answers is not in the document]
Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/join.py:24]: Done.

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/parse_json.py [hunk 2]
```diff
@@ -3,24 +3,34 @@
...
-        series = {}
-        for field in fields:
-            json_path = field.source.extract.json_path
-            if json_path is None:
-                continue
-            jsonpath_expression = jsonpath_rw.parse(json_path)
-            values = [match.value for match in jsonpath_expression.find(json)]
-            series[json_path] = values
-        return pd.DataFrame(series)
+def get_fields(record_set: RecordSet, graph: nx.MultiDiGraph) -> list[Field]:
```
[the remark this answers is not in the document]
Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/parse_json.py:13]: Good point! I delete the function on a coming PR where I have a new approach: now you can access fields with:

```python
record_set.fields
```

**Comment by marcenacp** [comment]
> I know this is already a big PR, but maybe we could consider adding some more guidance in the documentation as to how to specify FileProperty ? Currently we have only this: https://github.com/mlcommons/croissant/blob/main/docs/croissant-spec.md
> 
> but without any examples...

Done. I just added an example in docs/croissant-spec.md.

**Pull request merged**